src/main/java/org/thingsboard/gateway/extensions/mqtt/client/MqttBrokerMonitor.java

```diff
@@ -131,9 +139,38 @@ private void onDeviceData(List<DeviceData> data) {
             if (!dd.getTelemetry().isEmpty()) {
                 gateway.onDeviceTelemetry(dd.getName(), dd.getTelemetry());
             }
+            if (dd.isDisconnect()) {
+                log.debug("[{}] Will Topic Msg Received. Disconnecting device...", dd.getName());
+                gateway.onDeviceDisconnect(dd.getName());
```

Comment:
need to remove corresponding record in deviceKeepAliveTimers based on deviceName.